### PR TITLE
IGNITE-14901 Documents Discovery and SSL metrics.

### PR DESCRIPTION
**Pull request by ololo3000**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 1]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
 [cols="2,1,3",opts="header"]
 |===
 |Name|    Type|    Description
+|ActiveSessionsCount|   integer|   Active TCP sessions count.
 |OutboundMessagesQueueSize|   integer| Outbound messages queue size.
 |SentMessagesCount  | integer| Sent messages count.
 |SentBytesCount | long  |  Sent bytes count.
 |ReceivedBytesCount|  long|    Received bytes count.
 |ReceivedMessagesCount|   integer| Received messages count.
+|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to SSL errors (metric is exported only if SSL is enabled).
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:289]:
```suggestion
|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to the SSL errors (metric is exported only if SSL is enabled).
```

**Comment by IgGusev** [comment]
LGTM!

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 2]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+|SslEnabled|   boolean|   Whether SSL is enabled.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:290]:
```suggestion
|SslEnabled|   boolean|   Indicates whether SSL is enabled.
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 3]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+
+Register name: `connector.client.thin.tcp`
+
+
+[cols="2,1,3",opts="header"]
+|===
+|Name|    Type|    Description
+|ActiveSessionsCount|   integer|   Active TCP sessions count.
+|ReceivedBytesCount|   long|   Received bytes count.
+|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to SSL errors (metric is exported only if SSL is enabled).
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:305]:
```suggestion
|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to the SSL errors (metric is exported only if SSL is enabled).
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 4]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+|SentBytesCount|   long|   Sent bytes count.
+|SslEnabled|   boolean|   Whether SSL is enabled.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:307]:
```suggestion
|SslEnabled|   boolean|   Indicates whether SSL is enabled.
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 5]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+
+Register name: `connector.client.rest.tcp`
+
+
+[cols="2,1,3",opts="header"]
+|===
+|Name|    Type|    Description
+|ActiveSessionsCount|   integer|   Active TCP sessions count.
+|ReceivedBytesCount|   long|    Received bytes count.
+|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to SSL errors (metric is exported only if SSL is enabled).
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:322]:
```suggestion
|RejectedSslSessionsCount|   integer|   TCP sessions count that were rejected due to the SSL errors (metric is exported only if SSL is enabled).
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 6]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+|SentBytesCount|   long|   Sent bytes count.
+|SslEnabled|   boolean|   Whether SSL is enabled.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:324]:
```suggestion
|SslEnabled|   boolean|   Indicates whether SSL is enabled.
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 7]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+
+== Discovery IO
+
+Register name: `io.discovery`
+
+
+[cols="2,1,3",opts="header"]
+|===
+|Name|    Type|    Description
+|CoordinatorSince|   long|   Timestamp since which the local node was the coordinator (metric is only exported from server nodes).
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:337]:
```suggestion
|CoordinatorSince|   long|   Timestamp since which the local node was the coordinator (metric is exported only from server nodes).
```

Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:337]: I would rephrase this as follows:"...the local node **became** the coordinator...". Is it technically correct?

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 8]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+|Coordinator|   UUID|   Coordinator ID (metric is only exported from server nodes).
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:338]:
```suggestion
|Coordinator|   UUID|   Coordinator ID (metric is exported only from server nodes).
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 9]
```diff
@@ -280,11 +280,71 @@ Register name: `io.communication`
...
+|CurrentTopologyVersion|   long|   Current topology version.
+|JoinedNodes|   integer|   Joined nodes count.
+|LeftNodes|   integer|   Left nodes count.
+|MessageWorkerQueueSize|   integer|   Current message worker queue size.
+|PendingMessagesRegistered|   integer|   Pending registered messages count.
+|RejectedSslConnectionsCount|   integer|   TCP discovery connections count that were rejected due to SSL errors.
+|SslEnabled|   boolean|   Whether SSL is enabled.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:345]:
```suggestion
|RejectedSslConnectionsCount|   integer|   TCP discovery connections count that were rejected due to the SSL errors.
|SslEnabled|   boolean|   Indicates whether SSL is enabled.
```

**Review by Nikita-tech-writer — COMMENTED** [review]
@ololo3000 Please see my comments and suggested changes to the PR. Thank you.

File: docs/_docs/monitoring-metrics/new-metrics.adoc (hunk 7 above)
Comment (ololo3000) [reply docs/_docs/monitoring-metrics/new-metrics.adoc]: I think it's ok.

**Comment by ololo3000** [comment]
@Nikita-tech-writer I have fixed the comments. Could you please take another look?

**Review by NSAmelchev — APPROVED** [review]

**Comment by NSAmelchev** [comment]
LGTM
